openff/interchange/_annotations.py

```diff
@@ -151,6 +155,20 @@ def quantity_json_serializer(
     WrapSerializer(quantity_json_serializer),
 ]
 
+_MassQuantity = Annotated[
+    Quantity,
+    WrapValidator(quantity_validator),
+    AfterValidator(_is_mass),
+    WrapSerializer(quantity_json_serializer),
+]
+
```

Comment:
@Yoshanuikabundi these things (I don't know the right noun ...) appear to be re-usable as classes, not just annotations

1. Am I ... correct?
2. Should this magic live elsewhere, like in `openff-units`?
3. Should these things be "public" in some way?

Reply:
This is extremely cool. 

1. I think so - It looks like `Annotated[T]` is just a type alias for `T` that maintains a particular runtime value of the `__metadata__` attribute: https://github.com/python/cpython/blob/3.13/Lib/typing.py#L2210

2. I think for the time being putting this in `openff-models` would make the most sense. That way it can be used by all our Pydantical projects without being confused for a Pint thing. Since the validators would only kick in when the `Annotated[Quantity, ...]` is in a Pydantic model, I'd be wary of putting it in `openff-units`

3. I think having this be public in `openff-models` would be fine. I would love something like this to be public in `openff-units`, but the API there should be used something like:

```python
def foo(length: Quantity[NDArray, "angstrom"]):
   ...
```

I've taken a quick stab at this just now and it's a bit hairy but I think should be possible. In the interim, having this available for Pydantic projects would be great.